bastionzero/bzero
# Fix bug when SSHing as a nonexistent user

## Description of the change

See the [primary ticket](https://commonwealthcrypto.atlassian.net/browse/CWC-2192). The SSH logic was such that we weren't actually checking that the user has been successfully created before trying to set up that user's authorized keys file, causing a nil pointer dereference. 

So, fixed that -- both by checking for the error in the first place, and making it so that authorized keys simply returns an error when the user is nil instead of panicking). And also added another path on the daemon side to get errors in front of users.

## Testing

On develop:
  - A a user `baduser` to your target access policy
  - Run `zli generate sshConfig` if you need to
  - Try to `ssh baduser@<your-bzero-agent>` and it will crash!!

On this branch:
  - Same steps, but you will instead see `Failed to connect: the specified user does not exist on the target` (which is not perfect but it's a lot more specific than we used to be)

**backend branch:** 
**zli branch:** 

#### Ready to run system tests?


## Relevant release note information

Release Notes: Fix bug when SSHing as a nonexistent user

## Related JIRA tickets

Relates to JIRA: CWC-2065, CWC-2192

## Have you considered the security impacts?

Does this PR have any security impact?


If yes, please explain:
